tests/coprocessor/test_select.rs

```diff
@@ -103,6 +104,45 @@ impl Iterator for ChunkSpliter {
     }
 }
 
+struct DAGChunkSpliter {
+    chunks: Vec<Chunk>,
+    datums: Vec<Datum>,
+    col_cnt: usize,
+}
+
+impl DAGChunkSpliter {
+    fn new(chunks: Vec<Chunk>, col_cnt: usize) -> DAGChunkSpliter {
+        DAGChunkSpliter {
+            chunks: chunks,
+            col_cnt: col_cnt,
+            datums: Vec::with_capacity(0),
+        }
+    }
+}
+
+impl Iterator for DAGChunkSpliter {
+    type Item = Vec<Datum>;
+
+    fn next(&mut self) -> Option<Vec<Datum>> {
+        loop {
+            if self.chunks.is_empty() && self.datums.is_empty() {
+                return None;
+            } else if self.datums.is_empty() {
+                let chunk = self.chunks.swap_remove(0);
```

Comment:
self.chunks.pop().unwrap()?

Reply:
No, `pop()` returns the last element instead of first element of vector.